### PR TITLE
pythonPackages.jsbeautifier: 1.8.9 -> 1.9.1

### PR DESCRIPTION
**Pull request by apeyroux**
###### Motivation for this change

Up and "fix" #58608. I use doCheck = false; because there is a bug in the jsb test suite (cf #58608). It will have to be reactivated when the bug is fixed. I took the opportunity to upgrade.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/jsbeautifier/default.nix [hunk 1]
```diff
@@ -2,17 +2,19 @@
...
-  buildInputs = [ EditorConfig pytest ];
+  buildInputs = [ EditorConfig ];
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "7d02baa9b0459bf9c5407c1b99ad5566de04a3b664b18a58ac64f52832034204";
+    sha256 = "0q8ld072dkccssagjxyvc9633fb6ynflvz70924phgp3zxmim960";
   };
 
+  doCheck = false;
```
Comment (kalbasit) [inline pkgs/development/python-modules/jsbeautifier/default.nix]: let's not disable the tests, but fix them instead. The follow patch should make it work:

```patch
diff --git a/python/jsbeautifier/tests/test-packer.py b/python/jsbeautifier/tests/test-packer.py
index b2f8c790..3d7db760 100644
--- a/python/jsbeautifier/tests/test-packer.py
+++ b/python/jsbeautifier/tests/test-packer.py
@@ -8,7 +8,7 @@ global fails
 fails = 0
 
 
-def test_str(str, expected):
+def str_test(str, expected):
     global fails
     res = jsbeautifier.beautify(str, opts)
     if(res == expected):
@@ -23,17 +23,17 @@ def test_str(str, expected):
 str = "eval(function(p,a,c,k,e,d){e=function(c){return c.toString(36)};if(!''.replace(/^/,String)){while(c--){d[c.toString(a)]=k[c]||c.toString(a)}k=[function(e){return d[e]}];e=function(){return'\\w+'};c=1};while(c--){if(k[c]){p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c])}}return p}('2 0=\"4 3!\";2 1=0.5(/b/6);a.9(\"8\").7=1;',12,12,'str|n|var|W3Schools|Visit|search|i|innerHTML|demo|getElementById|document|w3Schools'.split('|'),0,{}))"
 expected = "var str = \"Visit W3Schools!\";\nvar n = str.search(/w3Schools/i);\ndocument.getElementById(\"demo\").innerHTML = n;"
 
-res = test_str(str, expected)
+res = str_test(str, expected)
 
 str = "a=b;\r\nwhile(1){\ng=h;{return'\\w+'};break;eval(function(p,a,c,k,e,d){e=function(c){return c.toString(36)};if(!''.replace(/^/,String)){while(c--){d[c.toString(a)]=k[c]||c.toString(a)}k=[function(e){return d[e]}];e=function(){return'\\w+'};c=1};while(c--){if(k[c]){p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c])}}return p}('$(5).4(3(){$(\'.1\').0(2);$(\'.6\').0(d);$(\'.7\').0(b);$(\'.a\').0(8);$(\'.9\').0(c)});',14,14,'html|r5e57|8080|function|ready|document|r1655|rc15b|8888|r39b0|r6ae9|3128|65309|80'.split('|'),0,{}))c=abx;"
 expected = "a = b;\nwhile (1) {\n    g = h; {\n        return '\\w+'\n    };\n    break;\n    $(document).ready(function() {\n        $('.r5e57').html(8080);\n        $('.r1655').html(80);\n        $('.rc15b').html(3128);\n        $('.r6ae9').html(8888);\n        $('.r39b0').html(65309)\n    });\n    c = abx;"
 
-res = test_str(str, expected)
+res = str_test(str, expected)
 
 str = "eval(function(p,a,c,k,e,r){e=function(c){return c.toString(36)};if('0'.replace(0,e)==0){while(c--)r[e(c)]=k[c];k=[function(e){return r[e]||e}];e=function(){return'[0-9ab]'};c=1};while(c--)if(k[c])p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c]);return p}('$(5).a(6(){ $(\'.8\').0(1); $(\'.b\').0(4); $(\'.9\').0(2); $(\'.7\').0(3)})',[],12,'html|52136|555|65103|8088|document|function|r542c|r8ce6|rb0de|ready|rfab0'.split('|'),0,{}))"
 expected = "$(document).ready(function() {\n    $(\'.r8ce6\').html(52136);\n    $(\'.rfab0\').html(8088);\n    $(\'.rb0de\').html(555);\n    $(\'.r542c\').html(65103)\n})"
 
-res = test_str(str, expected)
+res = str_test(str, expected)
 
 if (fails == 0):
     print("OK")
```

I made this patch by cloning the upstream repository and replicating the issue by running `nix run -f '<nixpkgs>' '(with import <nixpkgs> {}; python36.withPackages(ps: with ps; [ setuptools pytest ]))' -c python setup.py test` from within the python folder.

**Review by kalbasit — CHANGES_REQUESTED** [review]

**Comment by kalbasit** [comment]
I would also make a commit that fixes the tests and another for the update.

File: pkgs/development/python-modules/jsbeautifier/default.nix [hunk 2]
```diff
@@ -2,17 +2,19 @@
 
 buildPythonApplication rec {
   pname = "jsbeautifier";
-  version = "1.8.9";
+  version = "1.9.1";
 
   propagatedBuildInputs = [ six ];
 
-  buildInputs = [ EditorConfig pytest ];
+  buildInputs = [ EditorConfig ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/jsbeautifier/default.nix]: This belongs into `propagatedBuildInputs`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please correct the commit message.

File: pkgs/development/python-modules/jsbeautifier/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/jsbeautifier/default.nix]: EditorCOnfig should be in `propagatedBuildInputs`, pytest in `checkInputs`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/jsbeautifier/default.nix [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
   buildInputs = [ EditorConfig pytest ];
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "7d02baa9b0459bf9c5407c1b99ad5566de04a3b664b18a58ac64f52832034204";
+    sha256 = "0q8ld072dkccssagjxyvc9633fb6ynflvz70924phgp3zxmim960";
   };
 
+  patches = [(fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/beautify-web/js-beautify/pull/1654.diff";
```
[the remark this answers is not in the document]
Comment (kalbasit) [reply pkgs/development/python-modules/jsbeautifier/default.nix]: If I force push that PR, GitHub won't guarantee the commit will exist either. I prefer copying the patch locally and link to the PR in a comment above the patch.

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/jsbeautifier/default.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/jsbeautifier/default.nix]: I think it most probably will still exist sice GitHub PRs now have these "xyz force-pushed the foobar branch from hash to other_hash" messages.

[the remark this answers is not in the document]
Comment (apeyroux) [reply pkgs/development/python-modules/jsbeautifier/default.nix]: I used the commit rather than the pr because it is the most used with other packages.

**Comment by apeyroux** [comment]
@GrahamcOfBorg build pythonPackages.jsbeautifier

**Comment by apeyroux** [comment]
ping @FRidh

**Review by kalbasit — APPROVED** [review]

**Comment by kalbasit** [comment]
@GrahamcOfBorg build pythonPackages.jsbeautifier

File: pkgs/development/python-modules/jsbeautifier/default.nix [hunk 4]
```diff
@@ -1,18 +1,30 @@
...
 buildPythonApplication rec {
   pname = "jsbeautifier";
-  version = "1.8.9";
+  version = "1.9.1";
 
   propagatedBuildInputs = [ six ];
 
-  buildInputs = [ EditorConfig pytest ];
+  buildInputs = [ EditorConfig ];
+  checkInputys = [ pytest ];
```
Comment (kalbasit) [inline pkgs/development/python-modules/jsbeautifier/default.nix]:
```suggestion
  checkInputs = [ pytest ];
```

**Review by kalbasit — CHANGES_REQUESTED** [review]
The build is failing on Darwin.

EDIT: The build is failing because you are trying to modify a patch that was downloaded with fetchpatch. fetchpatch is just another derivation, and the $out of it is immutable.

File: pkgs/development/python-modules/jsbeautifier/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/development/python-modules/jsbeautifier/default.nix]: @apeyroux Why was this marked as resolved? EditorConfig is still not in propagatedBuildInputs.

Comment (apeyroux) [reply pkgs/development/python-modules/jsbeautifier/default.nix]: sorry ...

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.jsbeautifier python3.pkgs.jsbeautifier

File: pkgs/development/python-modules/jsbeautifier/default.nix [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/beautify-web/js-beautify/commit/78e35a11cbb805fc044241d6465800ee2bd57ebc.patch";
+      sha256 = "1ah7nshk96yljy37i20v4fga834dix9cdbhkdc3flfm4904n4523";
+    })
+  ];
+
+  prePatch = ''
+    substituteInPlace $patches --replace "/python/" "/"
```
Comment (dotlambda) [inline pkgs/development/python-modules/jsbeautifier/default.nix]: `patchFlags = [ "-p2" ]` instead of this might work

**Comment by kalbasit** [comment]
Please split the commit into 2 commits. One that fixes the build (it should apply to current version), and another for the update. Thanks!

**Comment by kalbasit** [comment]
@GrahamcOfBorg build python2.pkgs.jsbeautifier python3.pkgs.jsbeautifier

**Comment by flokli** [comment]
built and beautified some javascript, locally.

**Comment by flokli** [comment]
backported to 19.03 in dda301df365d36ee11ca434fd47bafea24df1674 and 971e5167e2c2ab0be1acd85617d4b96045298069.

**Pull request merged**